core/Object.php

```diff
@@ -816,7 +816,9 @@ protected function addMethodsFrom($property, $index = null) {
 		}
 
 		if(method_exists($extension, 'allMethodNames')) {
+			if (is_a($extension, 'Extension')) $extension->setOwner($this);
```

Comment:
Understandable. Same goes for the line below the allMethodNames() check.

I'll see if I have time to fix that monday and squash them together.